wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -183,43 +187,66 @@ class WirePlugin @Inject constructor(
     }
   }
 
+
   private fun mergeDependencyPaths(
     project: Project,
     dependencyPaths: Set<String>,
-    dependencyTrees: Set<SourceDirectorySet> = emptySet()
+    dependencyTrees: Set<SourceDirectorySet> = emptySet(),
+    dependencyJars: Set<ProtoRootSet> = emptySet()
   ): List<Any> {
     val allPaths = dependencyTrees.toMutableList<Any>()
 
     val parser = FileOrUriNotationConverter.parser()
+
+    dependencyJars.forEach { depJar ->
+      depJar.srcJar?.let { path ->
+        allPaths += mapDependencyPath(parser, path, project, depJar.includes)
+      }
+    }
+
     dependencyPaths.forEach { path ->
-      val converted = parser.parseNotation(path)
+      allPaths += mapDependencyPath(parser, path, project)
+    }
 
-      if (converted is File) {
-        val file =
-          if (!converted.isAbsolute) File(project.projectDir, converted.path) else converted
-        if (!file.exists()) {
-          throw IllegalArgumentException(
-              "Invalid path string: \"$path\". Path does not exist."
-          )
-        }
-        if (file.isDirectory) {
-          allPaths += project.files(path) as Any
-        } else {
-          throw IllegalArgumentException(
-              "Invalid path string: \"$path\". For individual files, use the closure syntax."
-          )
-        }
-      } else if (converted is URI && isURL(converted)) {
+    return allPaths
+  }
+
+  private fun mapDependencyPath(
+    parser: NotationParser<Any, Any>,
+    path: String,
+    project: Project,
+    jarIncludes: List<String> = emptyList()
+  ): Any {
+    val converted = parser.parseNotation(path)
+
+    if (converted is File) {
+      val file =
+        if (!converted.isAbsolute) File(project.projectDir, converted.path) else converted
+      if (!file.exists()) {
         throw IllegalArgumentException(
-            "Invalid path string: \"$path\". URL dependencies are not allowed."
+            "Invalid path string: \"$path\". Path does not exist."
+        )
+      }
+      return when {
+        file.isDirectory -> project.files(path)
+        file.isJar -> {
+          if (jarIncludes.isNotEmpty()) {
+            jarToIncludes[file.path] = jarIncludes
+          }
+          project.files(file)
+        }
+        else -> throw IllegalArgumentException(
+            "Invalid path string: \"$path\". For individual files, use the closure syntax."
```

Comment:
instead of saying "use the closure syntax" wanna just provide an example?

Reply:
I'll address in a follow-up, as this existed before.